sdks/go/test/integration/primitives/window_panes_test.go

```diff
@@ -26,3 +26,8 @@ func TestPanes(t *testing.T) {
 	integration.CheckFilters(t)
 	ptest.BuildAndRun(t, Panes)
 }
+
+func TestPanesNonStreaming(t *testing.T) {
```

Comment:
Just a drive-by: I'm uncertain how the different runners will treat panes in batch execution.

Triggers are more complicated than they appear, so thanks for giving this a go!

I know if they work, Flink will execute them correctly, regardless of Pipeline "mode" (batch or streaming), Dataflow might have different behavior for panes for Batch and Streaming pipelines.

Prism (the local runner), doesn't yet do anything with Panes, as it doesn't currently support triggers, which is the only material thing that mutates panes AFAICT.

My recommendation is making sure you get your expected behavior from Flink and Dataflow executions first, before trying to get Prism working (you might already be at this point, I see some changes to prism).

Do @mention me if you have questions!